### PR TITLE
MDEV-19508: SI_KERNEL not in solaris. Fixes AIX too

### PR DESCRIPTION
**Pull request by grooverdan**
Tested on openindiana Hipster 2019.10

I submit this under the MCA.

### CODE REVIEW

File: sql/mysqld.cc [hunk 1]
```diff
@@ -3287,7 +3287,11 @@ pthread_handler_t signal_hand(void *arg __attribute__((unused)))
       }
       break;
     case SIGHUP:
+#ifdef SI_KERNEL
       if (!abort_loop && origin != SI_KERNEL)
+#else
+      if (!abort_loop)
```
Comment (citrus-it) [inline sql/mysqld.cc]: For illumos, you probably want 
`if (!abort_loop && origin <= SI_USER)`
See https://github.com/omniosorg/illumos-omnios/blob/master/usr/src/uts/common/sys/siginfo.h#L124
- codes > 0 are from the kernel.

**Comment by robertbindar** [comment]
Hi @grooverdan, I tried to compile fresh 10.4 on latest openindiana hipster to test this patch and I get lots of other compile errors (sighandler_t errors, my_atomic_.* cast errors, etc), do you get those as well?
Also there is some ongoing work (discussions here: https://jira.mariadb.org/browse/MDEV-19508) to remove uses of `SI_KERNEL` and use `setsid` instead.

**Comment by grooverdan** [comment]
Like:
```
In file included from /export/home/dan/mariadb-server-10.4/mysys/my_lock.c:24:0:
/export/home/dan/mariadb-server-10.4/mysys/my_lock.c: In function 'my_lock':
/export/home/dan/mariadb-server-10.4/include/my_alarm.h:36:16: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
    alarm_signal=signal(SIGALRM,my_set_alarm_variable);
                ^
/export/home/dan/mariadb-server-10.4/mysys/my_lock.c:183:7: note: in expansion of macro 'ALARM_INIT'
       ALARM_INIT;
       ^~~~~~~~~~
/export/home/dan/mariadb-server-10.4/include/my_alarm.h:37:41: warning: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
 #define ALARM_END (void) signal(SIGALRM,alarm_signal); \
                                         ^
/export/home/dan/mariadb-server-10.4/mysys/my_lock.c:189:7: note: in expansion of macro 'ALARM_END'
       ALARM_END;
       ^~~~~~~~~
In file included from /usr/include/signal.h:42:0,
                 from /export/home/dan/mariadb-server-10.4/include/my_dbug.h:21,
                 from /export/home/dan/mariadb-server-10.4/include/my_global.h:1053,
                 from /export/home/dan/mariadb-server-10.4/mysys/mysys_priv.h:19,
                 from /export/home/dan/mariadb-server-10.4/mysys/my_lock.c:16:
/usr/include/iso/signal_iso.h:63:15: note: expected 'void (*)(int)' but argument is of type 'sig_return {aka void (*)(void)}'
 extern void (*signal(int, void (*)(int)))(int);
               ^~~~~~

```

And libmariadb requiring the same GSSAPI fix.

And possibly a couple of other errors that scrolled by too quick?

Yes.

Ack on MDEV-19508.

**Comment by grooverdan** [comment]
depends on #1491 (against 10.2) to compile right

**Comment by grooverdan** [comment]
so this leaves AIX with the bug of 07e9b1389857a8c0fee1dfda443ae257982fd2c9 for the time being until a `setsid` implementation is done, but at least it will and be functional.

**Comment by grooverdan** [comment]
Fixed.

**Pull request closed without merging**